### PR TITLE
Add export button back to reports, contacts, and messages

### PR DESCRIPTION
**Pull request by garethbowen**
#3113

### CODE REVIEW

File: static/js/services/generate-search-query.js [hunk 1]
```diff
@@ -1,140 +1,133 @@
...
+    var formatClinics = function(filters) {
+      var selected = getSelected(filters.facilities);
+      if (selected) {
+        return { clinic: selected };
+      }
+    };
+
+    var formatFreetext = function(filters) {
+      var freetext = filters.search;
+      if (freetext && freetext.indexOf(':') === -1) {
```
Comment (SCdF) [inline static/js/services/generate-search-query.js]: In the spirit of boyscouting, I think this line deserves a comment or function extraction with a more descriptive name. I'm guessing this has something to do with how you can search for `foo` and also `name:foo`, but I don't completely get it.

File: static/js/services/generate-search-query.js [hunk 2]
```diff
@@ -1,140 +1,133 @@
...
+          operands.push(formatClinics(filters));
+          operands.push(formatForm(filters));
+          operands.push(formatErrors(filters));
+          operands.push(formatVerified(filters));
         },
-        contacts: {
-          buildQuery: function(filters, operands) {
-            operands.push(formatFreetext(filters));
-            operands.push(formatContactsType(filters));
-            operands.push(formatClinics(filters));
```
Comment (SCdF) [inline static/js/services/generate-search-query.js]: [2] Did you mean to drop this?…

File: static/js/services/generate-search-query.js [hunk 3]
```diff
@@ -1,140 +1,133 @@
...
-
-      return function(typeName, filters) {
-        var type = TYPES[typeName];
-        if (!type) {
-          throw new Error('Unknown type');
+      },
+      contacts: {
+        buildQuery: function(filters, operands) {
+          operands.push(formatFreetext(filters));
+          operands.push(formatContactType());
```
Comment (SCdF) [inline static/js/services/generate-search-query.js]: [2] … ie shouldn't it be here?

File: templates/partials/actionbar.html [hunk 4]
```diff
@@ -62,46 +70,50 @@
...
+            <a class="mm-icon mm-icon-inverse mm-icon-caption" ng-show="actionBar.right.type === 'xml'" ui-sref="reports.edit({ reportId: actionBar.right.selected[0]._id })">
               <span class="fa fa-pencil"></span>
               <p translate>Edit</p>
             </a>
-            <a class="mm-icon mm-icon-inverse mm-icon-caption" ng-show="actionBar.type !== 'xml'" ng-click="emit('EditReport')">
+            <a class="mm-icon mm-icon-inverse mm-icon-caption" ng-show="actionBar.right.type !== 'xml'" ng-click="emit('EditReport')">
               <span class="fa fa-pencil"></span>
               <p translate>Update Facility</p>
             </a>
-            <a class="mm-icon mm-icon-inverse mm-icon-caption" ng-click="deleteDoc(actionBar.selected)" ng-class="{'mm-icon-disabled': actionBar.disableDelete}" mm-auth="can_delete_reports">
```
Comment (SCdF) [inline templates/partials/actionbar.html]: Why is the `ng-class` no longer valid?

File: static/js/services/generate-search-query.js [hunk 5]
```diff
@@ -1,140 +1,133 @@
...
+      }
+    };
+
+    var formatErrors = function(filters) {
+      if (filters.valid === true) {
+        return { errors: 0 };
+      }
+      if (filters.valid === false) {
         return {
-          type: selected || ContactSchema.getTypes()
```
Comment (garethbowen) [inline static/js/services/generate-search-query.js]: Yeah. We don't have a filter for contact types any more.

File: static/js/services/generate-search-query.js (hunk 3 above)
Comment (garethbowen) [inline static/js/services/generate-search-query.js]: I meant to drop it. We don't have a filter for clinics any more.

File: templates/partials/actionbar.html (hunk 4 above)
Comment (garethbowen) [inline templates/partials/actionbar.html]: `disableDelete` was never being set. Could find no reference in the code anywhere.

File: static/js/controllers/reports.js [hunk 6]
```diff
@@ -127,7 +128,7 @@ var _ = require('underscore'),
           model.type = doc.content_type;
```
Comment (garethbowen) [inline static/js/controllers/reports.js]: Changed to setRightActionBar

File: static/js/services/generate-search-query.js (hunk 1 above)
Comment (garethbowen) [inline static/js/services/generate-search-query.js]: Added a comment.

**Pull request merged**